src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,61 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final int DATE_INDEX = 0;
+    private static final int NAME_INDEX = 1;
+    private static final int WORK_HOURS_INDEX = 2;
+    private static final int PAY_PER_HOUR_INDEX = 3;
+    private static final String DATE_FORMAT = "dd.MM.yyyy";
+
+    public static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern(DATE_FORMAT);
+    private static final String DELIMITER = " ";
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate formatDateFrom = LocalDate.parse(dateFrom, formatter);
+        LocalDate formatDateTo = LocalDate.parse(dateTo, formatter);
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("Report for period ")
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo)
+                .append(System.lineSeparator());
+
+        salaryInfo(names, data, formatDateFrom, formatDateTo, stringBuilder);
```

Comment:
The method name 'salaryInfo' is not very informative. Consider renaming it to something more descriptive of its function, such as 'calculateAndAppendSalaries'.